src/Resizer.php

```diff
@@ -35,13 +38,23 @@ class Resizer implements ResizerInterface
     private $path;
 
     /**
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
      * {@inheritdoc}
      */
-    public function __construct(ResizeCalculatorInterface $calculator, Filesystem $filesystem, $path)
-    {
+    public function __construct(
+        ResizeCalculatorInterface $calculator,
+        Filesystem $filesystem,
+        $path,
+        EventDispatcher $eventDispatcher
```

Comment:
How about making the event dispatcher optional and set it via a `setEventDispatcher` method?